baselines/dasha/README.md

```diff
@@ -0,0 +1,124 @@
+
+
+## About this baseline
+
+**What’s implemented:** The code in this directory implements the experiments from the DASHA paper.
+
+**Datasets:** Mushrooms from LIBSVM and CIFAR10 from PyTorch's Torchvision
+
+**Hardware Setup:** These experiments were run on a desktop machine with 64 CPU cores. Any machine with 1 CPU would be able to run this code with the mushrooms dataset. The experiments with CIFAR10 would require slightly more CPU resources (e.g., 4 cores would be sufficient) and 1 GPU with CUDA.
+
+**Contributors:** Alexander Tyurin ([https://github.com/k3nfalt](https://github.com/k3nfalt))
+
+
+## Experimental Setup
+
+**Task:** Image Classification and Linear Regression
+
+**Model:** This baseline implements two models:
+
+* A logistic regression model with a nonconvex loss from the DASHA paper (Section A.1).
+* A neural network with the cross entropy loss (Section A.4).
+
+**Dataset:** This baseline only includes the MNIST dataset. By default, the datasets are partitioned randomly between $n$ clients:
+
+| Dataset | #classes | partitioning method |
+| :------ | :---: | :---: |
+| mushrooms | 2 | random |
+| CIFAR10 | 10 | random |
+
+**Training Hyperparameters:** In all experiments, we take parameters of algorithms predicted by the theory, except for the step sizes. In the case of the mushrooms's experiments, the step sizes are fine-tuned from the set of powers of two $\{0.25,0.5,1.0\}.$ In the case of CIFAR10's experiments, the step sizes are fixed to $0.01.$
+
+
+## Environment Setup
+
+To construct the Python environment follow these steps:
+
+```bash
+# Install the base Poetry environment
+# By default, Poetry installs the PyTorch package with Python 3.8 and CUDA 11.8.
+# If you have a different setup, then change the "torch" and "torchvision" lines in [tool.poetry.dependencies].
+poetry install
```

Comment:
We very recently changed the minimum version of python for baselines to 3.10. Could you reflect that in your comment above?
Also, i'm thinking on adding a complete description of how to setup the env using `pyenv`. This could be done by adding the following likes before your `poetry install`:

```
# Set Python 3.10
pyenv local 3.10.6
# Tell poetry to use python 3.10
poetry env use 3.10.6
# Now install your environment
```

See my other edit suggestion for the `pyproject.toml` to update to py3.10